src/components/SingleAutocompleteSelectField/SingleAutocompleteSelectField.tsx

```diff
@@ -298,12 +298,12 @@ const SingleAutocompleteSelectField: React.FC<SingleAutocompleteSelectFieldProps
     );
   }
 
+  const fuse = new Fuse(choices, { keys: ["label"] });
+
   return (
     <SingleAutocompleteSelectFieldComponent
       fetchChoices={q => setQuery(q || "")}
-      choices={filter(choices, query, {
-        key: "label",
-      })}
+      choices={query !== "" ? fuse.search(query).map(v => v.item) : choices}
```

Comment:
I will give you approval anyway, but if you have time, please consider now a more nice pattern - How about wrapping this into react hook, so we can re-use it across the other views. You can call the hook eg. `useFuzzySearch` that will create an instance of fusejs and share proper functions and results, encapsulating also the details like this map here.


```ts
const useFuzzySearch = <T>(items: T, keys: string[]) => {
  const fuse = new Fuse(items, { keys });
  
  const search = (query: string) => {
    if (!query) return items;
  
    return fuse.search(query)
  }
 
  return { search }
 }

```

Reply:
now we have an abstraction to it, and we can even replace fuzzy search library in one place

Reply:
> now we have an abstraction to it, and we can even replace fuzzy search library in one place

you are right. if needed, and used it other components, I will do it later.

Reply:
OK, do you want to change this or anything else now or we can merge it? (We'll do this abstraction anyway once we start migrating to fuse in the other places)

Reply:
what do you mean?

Reply:
I just want you to merge it.

Reply:
Shall I rebase with `main`?